test/controllers/calendar_controller_test.exs

```diff
@@ -0,0 +1,291 @@
+    offered_course_2 = insert(:offered_course, %{term: term, course: course2, students: [student1, student2], teachers: [teacher1, teacher2]})
+
+    classes =
+      [
+        insert(:class, %{offered_course: offered_course_1, date: "2017-01-01", starting_at: %{hour: 10, min: 0, sec: 0},  finishes_at: %{hour: 12, min: 0, sec: 0}, classroom: "r101"}),
+        insert(:class, %{offered_course: offered_course_1, date: "2017-01-02", starting_at: %{hour: 10, min: 0, sec: 0},  finishes_at: %{hour: 12, min: 0, sec: 0}, classroom: "r102"}),
+        insert(:class, %{offered_course: offered_course_1, date: "2017-01-03", starting_at: %{hour: 10, min: 0, sec: 0},  finishes_at: %{hour: 12, min: 0, sec: 0}, classroom: "r103"}),
+        insert(:class, %{offered_course: offered_course_1, date: "2017-01-04", starting_at: %{hour: 10, min: 0, sec: 0},  finishes_at: %{hour: 12, min: 0, sec: 0}, classroom: "r104"}),
+        insert(:class, %{offered_course: offered_course_2, date: "2017-01-05", starting_at: %{hour: 10, min: 0, sec: 0},  finishes_at: %{hour: 12, min: 0, sec: 0}, classroom: "r105"}),
+        insert(:class, %{offered_course: offered_course_1, date: to_string(Date.utc_today()), starting_at: %{hour: 10, min: 0, sec: 0},  finishes_at: %{hour: 12, min: 0, sec: 0}, classroom: "r106"})
+      ]
+
+    %{term: term,
+      courses: [course1, course2],
+      classes: classes,
+      students: [student1, student2],
+      teachers: [teacher1, teacher2],
+      offered_course_1: offered_course_1,
+      offered_course_2: offered_course_2,
+      teacher_with_no_class: teacher_with_no_class,
+      student_with_no_class: student_with_no_class}
+  end
+
+  setup(%{user_role: role}) do
+    user = insert(role)
+
+    conn =
+      Phoenix.ConnTest.build_conn()
+      |> assign(:current_user, user)
+    {:ok, conn: conn}
+  end
+
+  defp login_as(user) do
+    Phoenix.ConnTest.build_conn()
+    |> assign(:current_user, user)
+  end
+
+  describe "tests api with wrongly formatted parameters" do
+    @tag user_role: :coordinator
+    test "fails when date is not formatted correctly", %{conn: conn} do
+      params = %{date: "wrong date format", my_classes: "false"}
+      conn = get conn, calendar_path(conn, :index), params
+      assert json_response(conn, 406) == %{"error" => [%{"date" => "is invalid"}]}
+    end
+
+    @tag user_role: :coordinator
+    test "fails when my_classes is not a boolean", %{conn: conn} do
+      params = %{date: "2017-01-01", my_classes: "this is not a boolean"}
+      conn = get conn, calendar_path(conn, :index), params
+      assert json_response(conn, 406) == %{"error" => [%{"my_classes" => "is invalid"}]}
+    end
+
+    @tag user_role: :coordinator
+    test "fails and shows all errors when both params are not formatted correctly", %{conn: conn} do
+      params = %{date: "wrong date format", my_classes: "this is not a boolean"}
+      errors = %{"error" => [%{"date" => "is invalid"}, %{"my_classes" => "is invalid"}]}
```

Comment:
I put everything to Assign.errors so then i suggest one of the followings

1 - put "%{"errors" => %{"date" => "is invalid", "my_classes" => "is invalid"}}" in assign.errors but then what it shows will be %{"date" => "is invalid", "my_classes" => "is invalid"} 

2 - put them in another variable like assign.api_errors and then it can show 
%{"errors" => %{"date" => "is invalid", "my_classes" => "is invalid"}} when I return content of assign.api_error

@laszlokardinal-dina any opinion on this ?